File: python/source/interactive_books/domain/page_content.py
from dataclasses import dataclass

from interactive_books.domain.errors import BookError, BookErrorCode


@dataclass(frozen=True)
class PageContent:
    page_number: int
    text: str

    def __post_init__(self) -> None:
        if self.page_number < 1:
            raise BookError(
                BookErrorCode.PARSE_FAILED,
                f"Page number must be >= 1, got {self.page_number}",
            )

File: swift/InteractiveBooks/Sources/InteractiveBooksCore/Domain/ValueObjects/PageContent.swift
public struct PageContent: Sendable, Equatable {
    public let pageNumber: Int
    public let text: String

    public init(pageNumber: Int, text: String) throws {
        guard pageNumber >= 1 else {
            throw BookError.parseFailed(
                "Page number must be >= 1, got \(pageNumber)",
            )
        }
        self.pageNumber = pageNumber
        self.text = text
    }
}
